Library/Homebrew/cmd/link.rb

```diff
@@ -24,15 +24,17 @@ def link
     mode.dry_run = true if ARGV.dry_run?
 
     ARGV.kegs.each do |keg|
-      if HOMEBREW_PREFIX.to_s == "/usr/local" && keg.name == "openssl"
-        opoo <<-EOS.undent
-          Refusing to link: openssl
-          Linking keg-only OpenSSL means you may end up linking against the insecure,
-          deprecated system version while using the headers from the Homebrew version.
-          Instead, pass the full include/library paths to your compiler e.g.:
-            -I#{HOMEBREW_PREFIX}/opt/openssl/include -L#{HOMEBREW_PREFIX}/opt/openssl/lib
-        EOS
-        next
+      if HOMEBREW_PREFIX.to_s == "/usr/local"
+        if keg.name.start_with?("openssl") || keg.name.start_with?("libressl")
```

Comment:
> (My personal belief is that the prefix check could be completely dropped so that users get a more uniform behavior independently of where they chose to install Homebrew, though I do understand that there isn't any issue with non-/usr/local prefixes. Just a thought—I'm happy either way.)

I'd prefer we kept exempting non-`/usr/local` prefixes since there's not actually a problem there. I'd argue `openssl` shouldn't even really be `keg_only` outside of `/usr/local`, but there we go.

👍 on the rest of Martin's comment though.